src/analysis/complex/roots_of_unity.lean

```diff
@@ -0,0 +1,88 @@
+/-
+Copyright (c) 2020 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+
+import ring_theory.roots_of_unity
+import analysis.special_functions.trigonometric
+import analysis.special_functions.pow
+
+/-!
+# Complex roots of unity
+
+In this file we show that the `n`-th complex roots of unity
+are exactly the complex numbers `e ^ (2 * real.pi * complex.I * (i / n))` for `i ∈ finset.range n`.
+
+## Main declarations
+
+* `roots_of_unity_eq`: the complex `n`-th roots of unity are equal to the multiset of
+  complex numbers of the form `e ^ (2 * real.pi * complex.I * (i / n))` for `i ∈ finset.range n`.
+* `card_roots_of_unity`: the number of `n`-th roots of unity is exactly `n`.
+
+-/
+
+namespace complex
+
+open polynomial real
+
+/-- The complex `n`-th roots of unity are equal to the multiset of
+  complex numbers of the form `e ^ (2 * real.pi * complex.I * (i / n))` for `i ∈ finset.range n`. -/
+lemma roots_of_unity_eq (n : ℕ) :
+  roots_of_unity n ℂ =
+  multiset.map (λ (i : ℕ), exp (2 * pi * I * (i / n))) (multiset.range n) :=
+begin
+  by_cases hn : n = 0,
+  { simp only [hn, div_zero, roots_of_unity_zero, multiset.card_zero, nat.cast_zero,
+      multiset.map_const, multiset.range_zero, multiset.repeat_zero] },
+  have hn0 : (n : ℂ) ≠ 0, by assumption_mod_cast,
+  symmetry,
+  rw eq_iff_le_not_lt,
+  split,
+  { rw multiset.le_iff_subset,
+    { intro x,
+      simp only [multiset.mem_map, multiset.mem_range, mem_roots_of_unity (nat.pos_of_ne_zero hn)],
```

Comment:
Does this refer to the `mem_roots_of_unity` defined below? I'm a little surprised you can use these cyclically like this

Reply:
I think it's the one in `ring_theory/roots_of_unity`